pkg/datasource/csvsource/read_ticks.go

```diff
@@ -0,0 +1,55 @@
+package csvsource
+
+import (
+	"encoding/csv"
+	"io/fs"
+	"os"
+	"path/filepath"
+
+	"github.com/c9s/bbgo/pkg/types"
+)
+
+// TickReader is an interface for reading candlesticks.
+type TickReader interface {
+	Read(i int) (*CsvTick, error)
+	ReadAll(interval types.Interval) error
+}
```

Comment:
I like this interface. simple and clean.

Reply:
kuddos